### ISSUE TITLE
Crash when opening a case with an empty hand (or some blocks, depending on metadata)

### ISSUE THREAD

**Issue by RuinTD** [MightyPirates/OpenComputers#328]
Even though the title mentions two different conditions, they both produce the same error:

```
---- Minecraft Crash Report ----
// Hey, that tickles! Hehehe!

Time: 6/15/14 9:33 PM
Description: Unexpected error

java.lang.ArrayIndexOutOfBoundsException: -1
    at java.util.ArrayList.elementData(Unknown Source)
    at java.util.ArrayList.get(Unknown Source)
    at net.minecraftforge.oredict.OreDictionary.getOreName(OreDictionary.java:285)
    at li.cil.oc.util.Color$.isDye(Color.scala:52)
    at li.cil.oc.common.block.Delegator.func_149727_a(Delegator.scala:304)
    at net.minecraft.client.multiplayer.PlayerControllerMP.func_78760_a(PlayerControllerMP.java:326)
    at net.minecraft.client.Minecraft.func_147121_ag(Minecraft.java:1399)
    at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1903)
    at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:916)
    at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:835)
    at net.minecraft.client.main.Main.main(SourceFile:103)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:134)
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
    at java.util.ArrayList.elementData(Unknown Source)
    at java.util.ArrayList.get(Unknown Source)
    at net.minecraftforge.oredict.OreDictionary.getOreName(OreDictionary.java:285)
    at li.cil.oc.util.Color$.isDye(Color.scala:52)
    at li.cil.oc.common.block.Delegator.func_149727_a(Delegator.scala:304)
    at net.minecraft.client.multiplayer.PlayerControllerMP.func_78760_a(PlayerControllerMP.java:326)
    at net.minecraft.client.Minecraft.func_147121_ag(Minecraft.java:1399)

-- Affected level --
Details:
    Level name: MpServer
    All players: 1 total; [EntityClientPlayerMP['MCGamer20000'/11, l='MpServer', x=701.03, y=5.62, z=-772.16]]
    Chunk stats: MultiplayerChunkCache: 225, 225
    Level seed: 0
    Level generator: ID 01 - flat, ver 0. Features enabled: false
    Level generator options: 
    Level spawn location: World: (696,4,-766), Chunk: (at 8,0,2 in 43,-48; contains blocks 688,0,-768 to 703,255,-753), Region: (1,-2; contains chunks 32,-64 to 63,-33, blocks 512,0,-1024 to 1023,255,-513)
    Level time: 2484 game time, 2484 day time
    Level dimension: 0
    Level storage version: 0x00000 - Unknown?
    Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
    Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: false
    Forced entities: 77 total; [EntityCow['Cow'/16663, l='MpServer', x=738.75, y=4.00, z=-786.34], EntityCow['Cow'/16662, l='MpServer', x=738.50, y=4.00, z=-792.50], EntityCow['Cow'/16661, l='MpServer', x=739.41, y=4.00, z=-788.50], EntityPig['Pig'/16794, l='MpServer', x=714.50, y=4.00, z=-845.50], EntityPig['Pig'/16793, l='MpServer', x=717.50, y=4.00, z=-843.50], EntityChicken['Chicken'/16671, l='MpServer', x=733.34, y=4.00, z=-789.31], EntityChicken['Chicken'/16670, l='MpServer', x=736.56, y=4.00, z=-778.63], EntityChicken['Chicken'/16669, l='MpServer', x=736.44, y=4.00, z=-790.94], EntityPig['Pig'/16668, l='MpServer', x=740.50, y=4.00, z=-794.50], EntityPig['Pig'/16667, l='MpServer', x=740.50, y=4.00, z=-788.22], EntityClientPlayerMP['MCGamer20000'/11, l='MpServer', x=701.03, y=5.62, z=-772.16], EntityPig['Pig'/16666, l='MpServer', x=736.59, y=4.00, z=-794.75], EntityPig['Pig'/16665, l='MpServer', x=730.06, y=4.00, z=-794.09], EntityCow['Cow'/16664, l='MpServer', x=734.13, y=4.00, z=-786.97], EntityCow['Cow'/16559, l='MpServer', x=705.06, y=4.00, z=-819.03], EntityCow['Cow'/16558, l='MpServer', x=707.50, y=4.00, z=-811.50], EntityCow['Cow'/16557, l='MpServer', x=705.50, y=4.00, z=-809.50], EntitySheep['Sheep'/16556, l='MpServer', x=698.50, y=4.00, z=-810.50], EntitySheep['Sheep'/16555, l='MpServer', x=698.50, y=4.00, z=-806.50], EntitySheep['Sheep'/16554, l='MpServer', x=701.19, y=4.00, z=-808.19], EntitySheep['Sheep'/16553, l='MpServer', x=704.06, y=4.00, z=-805.13], EntityChicken['Chicken'/16672, l='MpServer', x=735.81, y=4.00, z=-790.38], EntityCow['Cow'/16568, l='MpServer', x=757.50, y=4.00, z=-692.50], EntityCow['Cow'/16566, l='MpServer', x=760.50, y=4.00, z=-692.50], EntityChicken['Chicken'/16564, l='MpServer', x=707.50, y=4.00, z=-808.50], EntityChicken['Chicken'/16562, l='MpServer', x=704.84, y=4.00, z=-805.25], EntityChicken['Chicken'/16563, l='MpServer', x=709.47, y=4.00, z=-817.47], EntityCow['Cow'/16560, l='MpServer', x=710.25, y=4.00, z=-816.50], EntityChicken['Chicken'/16561, l='MpServer', x=705.50, y=4.00, z=-806.50], EntityPig['Pig'/16585, l='MpServer', x=639.50, y=4.00, z=-809.03], EntityChicken['Chicken'/16584, l='MpServer', x=646.38, y=4.00, z=-811.53], EntityPig['Pig'/16587, l='MpServer', x=638.19, y=4.00, z=-811.81], EntityPig['Pig'/16586, l='MpServer', x=632.16, y=4.00, z=-817.16], EntityCow['Cow'/16577, l='MpServer', x=641.50, y=4.00, z=-809.50], EntityCow['Cow'/16579, l='MpServer', x=643.50, y=4.00, z=-806.50], EntityCow['Cow'/16578, l='MpServer', x=636.09, y=4.00, z=-802.94], EntityChicken['Chicken'/16581, l='MpServer', x=640.50, y=4.00, z=-812.50], EntityCow['Cow'/16580, l='MpServer', x=639.78, y=4.00, z=-810.59], EntityChicken['Chicken'/16583, l='MpServer', x=639.50, y=4.00, z=-812.50], EntityChicken['Chicken'/16582, l='MpServer', x=642.50, y=4.00, z=-812.50], EntityChicken['Chicken'/16601, l='MpServer', x=657.75, y=4.00, z=-840.56], EntityChicken['Chicken'/16602, l='MpServer', x=658.50, y=4.00, z=-840.50], EntityChicken['Chicken'/16603, l='MpServer', x=657.19, y=4.00, z=-840.44], EntityChicken['Chicken'/16710, l='MpServer', x=720.50, y=4.00, z=-843.50], EntityChicken['Chicken'/16604, l='MpServer', x=659.50, y=4.00, z=-842.50], EntityChicken['Chicken'/16711, l='MpServer', x=722.44, y=4.00, z=-847.38], EntityPig['Pig'/16605, l='MpServer', x=657.22, y=4.00, z=-841.50], EntityPig['Pig'/16606, l='MpServer', x=651.81, y=4.00, z=-842.91], EntityChicken['Chicken'/16709, l='MpServer', x=718.50, y=4.00, z=-839.50], EntityPig['Pig'/16607, l='MpServer', x=656.50, y=4.00, z=-833.50], EntityCow['Cow'/16714, l='MpServer', x=716.81, y=4.00, z=-837.19], EntityCow['Cow'/16715, l='MpServer', x=716.50, y=4.00, z=-834.50], EntityChicken['Chicken'/16712, l='MpServer', x=726.50, y=4.00, z=-841.50], EntityCow['Cow'/16713, l='MpServer', x=708.03, y=4.00, z=-830.13], EntityPig['Pig'/16718, l='MpServer', x=720.50, y=4.00, z=-846.50], EntityCow['Cow'/16716, l='MpServer', x=720.50, y=4.00, z=-834.50], EntityPig['Pig'/16717, l='MpServer', x=718.50, y=4.00, z=-841.50], EntityCow['Cow'/16617, l='MpServer', x=705.94, y=4.00, z=-723.84], EntityHorse['Horse'/16616, l='MpServer', x=713.50, y=4.00, z=-707.50], EntityCow['Cow'/16623, l='MpServer', x=704.56, y=4.00, z=-729.69], EntityCow['Cow'/16622, l='MpServer', x=704.81, y=4.00, z=-724.13], EntityCow['Cow'/16621, l='MpServer', x=705.41, y=4.00, z=-720.44], EntityCow['Cow'/16620, l='MpServer', x=696.25, y=4.00, z=-717.22], EntitySheep['Sheep'/16611, l='MpServer', x=658.41, y=4.00, z=-841.56], EntitySheep['Sheep'/16610, l='MpServer', x=659.03, y=4.00, z=-835.59], EntitySheep['Sheep'/16609, l='MpServer', x=650.13, y=4.00, z=-835.72], EntityPig['Pig'/16608, l='MpServer', x=656.50, y=4.00, z=-830.50], EntityHorse['Horse'/16615, l='MpServer', x=709.50, y=4.00, z=-708.50], EntityHorse['Horse'/16614, l='MpServer', x=707.13, y=4.00, z=-709.87], EntityHorse['Horse'/16613, l='MpServer', x=708.50, y=4.00, z=-713.50], EntitySheep['Sheep'/16612, l='MpServer', x=660.94, y=4.00, z=-836.06], EntityChicken['Chicken'/16626, l='MpServer', x=650.50, y=4.00, z=-693.50], EntityChicken['Chicken'/16627, l='MpServer', x=647.50, y=4.00, z=-695.50], EntityCow['Cow'/16624, l='MpServer', x=712.75, y=4.00, z=-730.25], EntityChicken['Chicken'/16630, l='MpServer', x=646.50, y=4.00, z=-692.50], EntityChicken['Chicken'/16628, l='MpServer', x=644.50, y=4.00, z=-693.50], EntityChicken['Chicken'/16629, l='MpServer', x=647.50, y=4.00, z=-693.50]]
    Retry entities: 0 total; []
    Server brand: fml,forge
    Server type: Integrated singleplayer server
Stacktrace:
    at net.minecraft.client.multiplayer.WorldClient.func_72914_a(WorldClient.java:368)
    at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:2382)
    at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:864)
    at net.minecraft.client.main.Main.main(SourceFile:103)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:134)
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
    Minecraft Version: 1.7.2
    Operating System: Windows 8.1 (amd64) version 6.3
    Java Version: 1.7.0_55, Oracle Corporation
    Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
    Memory: 236472232 bytes (225 MB) / 519045120 bytes (495 MB) up to 954728448 bytes (910 MB)
    JVM Flags: 2 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx1G
    AABB Pool Size: 3729 (208824 bytes; 0 MB) allocated, 3 (168 bytes; 0 MB) used
    IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
    FML: MCP v9.03 FML v7.2.211.1121 Minecraft Forge 10.12.2.1121 6 mods loaded, 6 mods active
    mcp{9.03} [Minecraft Coder Pack] (minecraft.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
    FML{7.2.211.1121} [Forge Mod Loader] (forge-1.7.2-10.12.2.1121.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
    Forge{10.12.2.1121} [Minecraft Forge] (forge-1.7.2-10.12.2.1121.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
    OpenComponents{0.1.0.2} [OpenComponents] (OpenComponents-MC1.7.2-0.1.1.6-universal.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
    OpenComputers|Core{1.0.0} [OpenComputers (Core)] (OpenComputers-MC1.7.2-1.2.12.144-universal.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
    OpenComputers{1.2.12.144} [OpenComputers] (OpenComputers-MC1.7.2-1.2.12.144-universal.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
    Launched Version: 1.7.2-Forge10.12.2.1121
    LWJGL: 2.9.0
    OpenGL: Intel(R) HD Graphics GL version 3.1.0 - Build 9.17.10.3347, Intel
    Is Modded: Definitely; Client brand changed to 'fml,forge'
    Type: Client (map_client.txt)
    Resource Packs: []
    Current Language: English (US)
    Profiler Position: N/A (disabled)
    Vec3 Pool Size: 380 (21280 bytes; 0 MB) allocated, 21 (1176 bytes; 0 MB) used
    Anisotropic Filtering: Off (1)
```

**Comment by RuinTD** [comment]
I just noticed this is (maybe) already fixed in 9d430b1869affe5e8444e47f56e1a55a5f64232e, but shouldn't there be a hotfix for this or something?

**Comment by fnuecke** [comment]
Yeah, it's fixed in 1.3 already. I wanted to finish cleaning that up to release a proper beta, I'll look into backporting a couple of fixes (including this one) to release a 1.2 final for those that can't update to 1.3 due to dependencies.

**Comment by Wuerfel21** [comment]
@fnuecke this release should contain the screenfix

**Comment by fnuecke** [comment]
Should be fixed in the latest 1.2 builds (http://ci.cil.li/ the OpenComputers-1.2-MC\* ones).
